File: app/src/main/java/com/bookcrossing/mobile/modules/LocationModule.java
package com.bookcrossing.mobile.modules;

import com.google.android.gms.location.FusedLocationProviderClient;
import com.google.android.gms.location.LocationServices;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import io.nlopez.smartlocation.SmartLocation;

/**
 * (c) 2017 Andrey Mukamolov <fobo66@protonmail.com>
 * Created 6/25/17.
 */

@Module public class LocationModule {
  @Provides @Singleton public SmartLocation provideSmartLocation(App app) {
    return SmartLocation.with(app.getApplicationContext());
  }

  @Provides
  @Singleton
  public FusedLocationProviderClient provideFusedLocationProviderClient(App app) {
    return LocationServices.getFusedLocationProviderClient(app.getApplicationContext());
  }
}

File: app/src/main/java/com/bookcrossing/mobile/modules/LocationModule.kt
package com.bookcrossing.mobile.modules

import android.content.Context
import com.google.android.gms.location.FusedLocationProviderClient
import com.google.android.gms.location.LocationServices
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * (c) 2017 Andrey Mukamolov <fobo66@protonmail.com>
 * Created 6/25/17.
 */

@Module
class LocationModule {
  @Provides
  @Singleton
  fun provideFusedLocationProviderClient(context: Context): FusedLocationProviderClient {
    return LocationServices.getFusedLocationProviderClient(context)
  }
}
